packages/sanity/src/core/form/inputs/PortableText/object/BlockObject.tsx

```diff
@@ -76,29 +85,52 @@ export function BlockObject(props: BlockObjectProps) {
   const markers = usePortableTextMarkers(path)
   const editor = usePortableTextEditor()
   const memberItem = usePortableTextMemberItem(pathToString(path))
+  const isDeleting = useRef<boolean>(false)
+
+  const selfSelection = useMemo(
+    (): EditorSelection => ({
+      anchor: {path: relativePath, offset: 0},
+      focus: {path: relativePath, offset: 0},
+    }),
+    [relativePath]
+  )
 
   const handleMouseOver = useCallback(() => setReviewChangesHovered(true), [])
   const handleMouseOut = useCallback(() => setReviewChangesHovered(false), [])
 
   const onOpen = useCallback(() => {
-    PortableTextEditor.blur(editor)
-    onItemOpen(path)
-  }, [editor, onItemOpen, path])
+    if (memberItem) {
+      PortableTextEditor.blur(editor)
+      onItemOpen(memberItem.node.path)
+    }
+  }, [editor, onItemOpen, memberItem])
 
   const onClose = useCallback(() => {
     onItemClose()
+    PortableTextEditor.select(editor, selfSelection)
     PortableTextEditor.focus(editor)
-  }, [editor, onItemClose])
+  }, [onItemClose, editor, selfSelection])
 
   const onRemove = useCallback(() => {
-    const sel: EditorSelection = {
-      focus: {path: relativePath, offset: 0},
-      anchor: {path: relativePath, offset: 0},
+    // Guard against clicking "Delete" multiple times.
+    if (isDeleting.current) {
```

Comment:
I'm not sure if this is strictly necessary to gard like this here, but I'm starting to think it's a nice pattern. Maybe more important if the object was created.

Reply:
Yes, that's a good idea. Adding it.